build/cpp/kvstore/src/kvstore_test.cc

```diff
@@ -20,15 +19,68 @@ TEST(KVStore, PuttingMultiple) {
   EXPECT_EQ((*kvstore.Get("key1"))[2], "value3");
 }
 
-// test put then remove
+// put then remove
 TEST(KVStore, PutAndRemove) {
-  kvstore::KVStore kvstore;
+  kvstore::KVStore kvstore("");
   kvstore.Put("key1", "value1");
   EXPECT_EQ((*kvstore.Get("key1"))[0], "value1");
   kvstore.Remove("key1");
   EXPECT_EQ(kvstore.Get("key1"), std::nullopt);
 }
 
+// write file on existing file
+TEST(KVStore, ExistingWriteFile) {
+  kvstore::KVStore kvstore("test1.txt");
+  kvstore.Put("key1", "value1");
+  kvstore.Put("key1", "value2");
+  kvstore.Put("key2", "value3");
+  kvstore.WriteFile();
+  kvstore.WriteFile();
+  }
+
+// write file on non-existent file
+TEST(KVStore, WriteFileNonExistent) {
+  kvstore::KVStore kvstore("test2.txt");
+  kvstore.Put("key1", "value1");
+  kvstore.Put("key1", "value2");
+  kvstore.Put("key2", "value3");
+  kvstore.WriteFile();
+}
+
+// write file on unknown file format
+TEST(KVStore, WriteFileUnknownFileFormat) {
+  kvstore::KVStore kvstore("test3.test3");
+  kvstore.Put("key1", "value1");
+  kvstore.Put("key1", "value2");
+  kvstore.Put("key2", "value3");
+  kvstore.WriteFile();
+}
+
+// read file on existing file
+TEST(KVStore, ReadFileExistent) {
+  kvstore::KVStore kvstore("test1.txt");
+  kvstore.ReadFile();
+  EXPECT_EQ((*kvstore.Get("key1"))[0], "value1");
+  EXPECT_EQ((*kvstore.Get("key1"))[1], "value2");
+  EXPECT_EQ((*kvstore.Get("key2"))[0], "value3");
+}
+
+// read file on non-existent file
+TEST(KVSTORE, ReadFileNonExistent) {
+  kvstore::KVStore kvstore("test4.txt");
+  kvstore.ReadFile();
+  EXPECT_EQ(kvstore.Get("key1"), std::nullopt);
+}
+
+// read file on unknown file format
```

Comment:
Unsure whether this test is necessary. If the .test3 files and .txt files are formatted the same way then doesn't this test the same thing as ReadFileExistent?

Reply:
This test was removed.